### PR TITLE
add ceph-poc

### PR DESCRIPTION
**Pull request by toshipp**
add a PoC rook/ceph cluster to evaluate RGW performance improvement with SSD using the index.

### CODE REVIEW

**Review by llamerada-jp — APPROVED** [review]
LGTM

**Review by llamerada-jp — DISMISSED** [review]
LGTM after squash

**Review by satoru-takeuchi — CHANGES_REQUESTED** [review]

File: rook/overlays/gcp/kustomization.yaml [hunk 1]
```diff
@@ -32,6 +45,34 @@ patchesJson6902:
...
+      value: 3
+    - op: replace
+      path: /spec/storage/storageClassDeviceSets/0/volumeClaimTemplates/0/spec/resources/requests/storage
+      value: 5Gi
+    - op: replace
+      path: /spec/storage/storageClassDeviceSets/1/count
+      value: 3
+    - op: replace
+      path: /spec/storage/storageClassDeviceSets/1/volumeClaimTemplates/0/spec/resources/requests/storage
+      value: 3Gi
```
[the remark this answers is not in the document]
Comment (toshipp) [reply rook/overlays/gcp/kustomization.yaml:75]: I think it is a copy miss, but I can not find the requirement. Would you show me that?

[the remark this answers is not in the document]
Comment (satoru-takeuchi) [reply rook/overlays/gcp/kustomization.yaml:75]: OK, here it is.

https://docs.ceph.com/en/latest/cephadm/osd/
> The device must be larger than 5 GB.

[the remark this answers is not in the document]
Comment (toshipp) [reply rook/overlays/gcp/kustomization.yaml:75]: Thanks!
BTW, ceph says the osd is available even when the disk is lower than 5GB 🤔

[the remark this answers is not in the document]
Comment (toshipp) [reply rook/overlays/gcp/kustomization.yaml:75]: I'll squash a new commit.

[the remark this answers is not in the document]
Comment (satoru-takeuchi) [reply rook/overlays/gcp/kustomization.yaml]: Hmm, it's odd. IIRC, ceph-volume fails if the device is less than 5G.

I guess the reason is the difference between requested size written in PVC and the actual size of volume.  In our HDD cluster, PVC is always backed by whole device.

As for OSD on SSD, one possibility is that  topolvm creats at least 5GB volume or so. @llamerada-jp Do you know anything?

[the remark this answers is not in the document]
Comment (satoru-takeuchi) [reply rook/overlays/gcp/kustomization.yaml]: Anyway I believe the actual volume size >= 5G.

**Review by llamerada-jp — APPROVED** [review]
LGTM

File: rook/overlays/gcp/kustomization.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (toshipp) [reply rook/overlays/gcp/kustomization.yaml]: In my test environment, the size is actually 3GB and the odd pod works fine.

```
cybozu@gcp0-boot-0:~$ kubectl exec -it  -n ceph-poc  rook-ceph-osd-3-64f76c449b-dbqkw -- blockdev  --getsize64 /var/lib/ceph/osd/ceph-3/block
3221225472
cybozu@gcp0-boot-0:~$ kubectl get pod   -n ceph-poc  rook-ceph-osd-3-64f76c449b-dbqkw
NAME                               READY   STATUS    RESTARTS   AGE
rook-ceph-osd-3-64f76c449b-dbqkw   1/1     Running   0          41h
```

BTW, this is not related to this PR, would you accept this? If not, we can not merge this...

[the remark this answers is not in the document]
Comment (satoru-takeuchi) [reply rook/overlays/gcp/kustomization.yaml]: Sure.

**Review by satoru-takeuchi — APPROVED** [review]

**Comment by satoru-takeuchi** [comment]
Please merge this PR after the integration test gets green.

File: rook/overlays/gcp/kustomization.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (toshipp) [reply rook/overlays/gcp/kustomization.yaml]: 🙇

**Pull request merged**